boomonkeyboo/scratch_project
# Wg/fixed update user newpassword issue

Issue Type
☑︎ Bug
☐ Feature
☐ Tech Debt

**Description**
Fixed user profile change no password change problem proposed by Caleb.

**Steps to Validate Feature**
Run npm run dev to start server
postman PATCH request to localhost:3001/api/updateuser, 
Use 
{ "userid":"3b631f15-55ef-4e25-a63e-10a59fabd2ba", 
"newfn" : "wenzhen_new", 
"newln": "gong_new", 
"newemail": "wzg_new@gmail.com", 
"newusername": "wenzhengong_new"
}
as req.body
Send request. New user object should be returned. Password not changed.

Again use
{ "userid":"3b631f15-55ef-4e25-a63e-10a59fabd2ba", 
"newfn" : "wenzhen_new", 
"newln": "gong_new", 
"newemail": "wzg_new@gmail.com", 
"newusername": "wenzhengong_new",
"newpassword": "newpassword"
}
as req.body
Send request. New user object should be returned. Password now changed.

**Previous behavior**
hen user does not want to change password,  according to the previous logic,  the destructured newpassword variable will be a hash from old password and when it gets Bcrypted again, it will be a different hash.

**Expected behavior**
When user update their profile, they can choose to leave out password field or not. Server will respond to whether they change old password or not accordingly.

**Screenshots & Videos**
N/A

**Additional context**
Credit to Tristan, added undefined to CORS whitelist and now frontend and postman can both send request.
